File: clients/client-c/legacy_calls.py
"""Fake client service C — calls legacy order endpoints."""

import os
import urllib.request

LEGACY_API_BASE = os.environ.get("LEGACY_API_BASE", "http://localhost:8080")


def fetch(path: str) -> None:
    with urllib.request.urlopen(f"{LEGACY_API_BASE}{path}") as response:
        response.read()


fetch("/orders")
fetch("/orders/5/status")
fetch("/customers/3/orders")

File: clients/client-b/legacy_calls.mjs
/**
 * Fake client service B — calls legacy customer, product, and order endpoints.
 * Used by the Phase 1 discovery agent to build inventory.json.
 */

const LEGACY_API_BASE = process.env.LEGACY_API_BASE ?? "http://localhost:8080";

async function fetchLegacy(path, options = {}) {
  const response = await fetch(`${LEGACY_API_BASE}${path}`, options);
  return response.json();
}

await fetchLegacy("/products/2");
await fetchLegacy("/customers");
await fetchLegacy("/customers/2");
await fetchLegacy("/orders/3", { method: "GET" });
